chained/algeneva_impl.go

```diff
@@ -0,0 +1,59 @@
+package chained
+
+import (
+	"context"
+	"net"
+
+	"github.com/getlantern/common/config"
+	algeneva "github.com/getlantern/lantern-algeneva"
+	"github.com/getlantern/netx"
+
+	"github.com/getlantern/flashlight/v7/ops"
+)
+
+type algenevaImpl struct {
+	nopCloser
+	dialerOps      algeneva.DialerOpts
+	addr           string
+	reportDialCore reportDialCoreFn
+}
+
+func newAlgenevaImpl(addr string, pc *config.ProxyConfig, reportDialCore reportDialCoreFn) (*algenevaImpl, error) {
+	strategy := ptSetting(pc, "algeneva_strategy")
+
+	ops := algeneva.DialerOpts{
+		AlgenevaStrategy: strategy,
+	}
+	return &algenevaImpl{
+		dialerOps:      ops,
+		addr:           addr,
+		reportDialCore: reportDialCore,
+	}, nil
+}
+
+func (a *algenevaImpl) dialServer(op *ops.Op, ctx context.Context) (net.Conn, error) {
+	dialerOps := a.dialerOps
+	dialerOps.Dialer = &algenevaDialer{
+		a.reportDialCore,
+		op,
+	}
+
+	return algeneva.DialContext(ctx, "TCP", a.addr, dialerOps)
```

Comment:
This needs to be lowercase: "tcp".  Otherwise, [you get an error](https://go.dev/play/p/QOlbBoxDKA5): `dial TCP: unknown network TCP`.

Reply:
That's right..oops..